devices/YarpRobotLoggerDevice/include/BipedalLocomotion/YarpRobotLoggerDevice.h

```diff
@@ -193,14 +197,20 @@ class YarpRobotLoggerDevice : public yarp::dev::DeviceDriver,
                     std::size_t vectorSize,
                     const std::vector<std::string>& metadata = {});
 
+    bool populateCamerasData(const std::string& logPrefix,
+                             std::shared_ptr<const ParametersHandler::IParametersHandler> params,
+                             const std::string& fpsParamName,
+                             const std::vector<std::string>& cameraNames);
+
     bool hasSubstring(const std::string& str, const std::vector<std::string>& substrings) const;
     void recordVideo(const std::string& cameraName, VideoWriter& writer);
+    void saveCodeStatus(const std::string& logPrefix, const std::string& fileName) const;
     void unpackIMU(Eigen::Ref<const analog_sensor_t> signal,
                    Eigen::Ref<accelerometer_t> accelerometer,
                    Eigen::Ref<gyro_t> gyro,
                    Eigen::Ref<orientation_t> orientation);
     bool setupRobotSensorBridge(std::weak_ptr<const ParametersHandler::IParametersHandler> params);
-    bool setupRobotCameraBridge(std::weak_ptr<const ParametersHandler::IParametersHandler> params);
+    bool setupRobotCameraBridge(std::shared_ptr<const ParametersHandler::IParametersHandler> params);
```

Comment:
I don't think this is required 🤔

Reply:
Forget the comment above, I checked better and I can revert it 😁

Reply:
Fixed in https://github.com/ami-iit/bipedal-locomotion-framework/pull/989/commits/f01135f336b707cec430f1c0df618bd421961e73